src/twister2/platform_specification.py

```diff
@@ -113,12 +116,47 @@ class PlatformSchema(Schema):
 
 def discover_platforms(directory: Path) -> Generator[PlatformSpecification, None, None]:
     """Return platforms from given directory"""
-    for file in directory.glob('*/*/*.yaml'):
+    for spec_file in directory.glob('*/*/*.yaml'):
         try:
-            yield PlatformSpecification.load_from_yaml(str(file))
+            platform = PlatformSpecification.load_from_yaml(str(spec_file))
         except Exception as e:
             logger.exception('Cannot read platform definition from yaml: %s', e)
             raise
+        yield platform
+        if '@' not in platform.identifier:
+            yield from discover_platform_revisions(platform, spec_file.parent)
+
+
+def discover_platform_revisions(
+    platform: PlatformSpecification, directory: Path
+) -> Generator[PlatformSpecification, None, None]:
+    """
+    Return all revisions for platform specification in the directory.
+
+    :param platform: platform specification
+    :param directory: directory to search revision in
+    :return: all revisions for given platform
+    """
+    revision_patterns: list[str] = [
+        '[A-Z]',
+        '[0-9]+',
+        '(0|[1-9][0-9]*)(_[0-9]+)*(_[0-9]+)*'
+    ]
+    files_list: list[str] = os.listdir(directory)
+    for file in files_list:
+        # Need to make sure the revision matches
+        # the permitted patterns as described in
+        # cmake/modules/extensions.cmake.
+        for pattern in revision_patterns:
+            if match := re.match(f'{platform.identifier}_(?P<revision>{pattern})\\.conf', file):
+                revision: str = match.group('revision')
+                filename = f'{platform.identifier}_{revision}.yaml'
+                if filename not in files_list:
+                    platform_revision = copy.deepcopy(platform)
+                    revision = revision.replace('_', '.')
+                    platform_revision.identifier = f'{platform.identifier}@{revision}'
+                    platform_revision.testing.default = False
+                    yield platform_revision
```

Comment:
`break` missed here (compared to Twister V1 implementation). 
File like `qemu_x86_tiny_768.conf` is matched two times by `revision_patterns = '[0-9]+', '(0|[1-9][0-9]*)(_[0-9]+)*(_[0-9]+)*'`

Reply:
Fixed and added unit tests. 
I took same regex patterns as V1, it seems they are wrong, but somehow works in V1.

Reply:
It works in TwisterV1 because of `break` in for loop after match:
https://github.com/zephyrproject-rtos/zephyr/blob/5786b63e6ca9b64f2e27308564f0a8737bd4586f/scripts/pylib/twister/twisterlib/testplan.py#L398
this is the difference